src/vm/jitinterface.cpp

```diff
@@ -6631,6 +6632,12 @@ const char* CEEInfo::getMethodNameFromMetadata(CORINFO_METHOD_HANDLE ftnHnd, con
         {
             MethodTable* pMT = ftn->GetMethodTable();
             classResult = pMT->GetFullyQualifiedNameInfo(&namespaceResult);
+            // Query enclosingClassName when it is required
+            // The namespace of nested classes is empty
+            if (pMT->GetClass()->IsNested() && enclosingClassName != NULL)
```

Comment:
It is weird that this would return different namespace depending on whether you pass in null or non-null `enclosingClassName`. I would delete the `enclosingClassName` check here.